graphsync.go

```diff
@@ -5,18 +5,45 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/google/uuid"
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/datamodel"
 	"github.com/ipld/go-ipld-prime/traversal"
 	"github.com/libp2p/go-libp2p-core/peer"
 )
 
 // RequestID is a unique identifier for a GraphSync request.
-type RequestID int32
+type RequestID struct{ string }
```

Comment:
I experimented with a few things here but landed on this after copying `cid.CID`'s pattern with the main reasoning being that this is really a byte string, I don't want users to consider it reasonable to cast this to a user-readable `string` (knowing it's a UUID and seeing it's a typed `string`, that would be a reasonable API assumption I reckon), they really should `.String()` it to get the string form of a UUID. The same thing holds for `CID` (and probably should for other types that are not utf8 strings IMO).

And, I didn't use `[]byte` because it needs to be used for map keys (I learnt my Go lesson about that while trying it!).

Happy to change it if you think it should be a simple `string` though.